installation/scripts/prom-mtls-patch.sh

```diff
@@ -0,0 +1,405 @@
+  volumes:
+    - emptyDir:
+        medium: Memory
+      name: istio-certs
+  volumeMounts:
+    - mountPath: /etc/prometheus/secrets/istio.default/
+      name: istio-certs
+EOF
+  )
+
+  echo "${patch}" > patch.yaml
+  kubectl apply -f patch.yaml
+  rm patch.yaml
+}
+
+function patchAlertManagerForMTLS() {
+  patch=$(cat <<"EOF"
+apiVersion: monitoring.coreos.com/v1
+kind: Alertmanager
+metadata:
+  name: monitoring-alertmanager
+  namespace: kyma-system
+spec:
+  podMetadata:
+    annotations:
+      sidecar.istio.io/inject: "true"
+EOF
+  )
+
+  echo "${patch}" > patch.yaml
+  kubectl apply -f patch.yaml
+  rm patch.yaml
+}
+
+function patchDeploymentsToInjectSidecar() {
+  allDeploy=(
+    kiali-server
+    monitoring-kube-state-metrics
+    monitoring-operator
+    api-gateway
+  )
+
+  resource="deployment"
+  namespace="kyma-system"
+
+  for depl in "${allDeploy[@]}"; do
+    if kubectl get ${resource} -n ${namespace} "${depl}" > /dev/null; then
+      kubectl get ${resource} -n ${namespace} "${depl}" -o yaml > "${depl}.yaml"
+
+      if [[ "$OSTYPE" == "darwin"* ]]; then
+        sed -i '' -e 's/sidecar.istio.io\/inject: "false"/sidecar.istio.io\/inject: "true"/g' "${depl}.yaml"
+      else # assume Linux otherwise
+        sed -i 's/sidecar.istio.io\/inject: "false"/sidecar.istio.io\/inject: "true"/g' "${depl}.yaml"
+      fi
+
+      kubectl apply -f "${depl}.yaml" || true
+
+      rm "${depl}.yaml"
+    fi
+  done
+}
+
+function enableNodeExporterMTLS() {
+  monitor=$(cat <<"EOF"
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  annotations:
+    meta.helm.sh/release-name: monitoring
+    meta.helm.sh/release-namespace: kyma-system
+  labels:
+    app: monitoring-node-exporter
+    app.kubernetes.io/instance: monitoring
+    app.kubernetes.io/managed-by: Helm
+    app.kubernetes.io/name: monitoring
+    chart: monitoring-1.0.0
+    helm.sh/chart: monitoring-1.0.0
+    release: monitoring
+  name: monitoring-node-exporter
+  namespace: kyma-system
+spec:
+  endpoints:
+  - metricRelabelings:
+    - action: keep
+      regex: ^(go_goroutines|go_memstats_alloc_bytes|go_memstats_heap_alloc_bytes|go_memstats_heap_inuse_bytes|go_memstats_heap_sys_bytes|go_memstats_stack_inuse_bytes|node_.*|process_cpu_seconds_total|process_max_fds|process_open_fds|process_resident_memory_bytes|process_start_time_seconds|process_virtual_memory_bytes)$
+      sourceLabels:
+      - __name__
+    port: metrics
+    scheme: https
+    tlsConfig:
+      caFile: /etc/prometheus/secrets/istio.default/root-cert.pem
+      certFile: /etc/prometheus/secrets/istio.default/cert-chain.pem
+      keyFile: /etc/prometheus/secrets/istio.default/key.pem
+      insecureSkipVerify: true
+  jobLabel: jobLabel
+  selector:
+    matchLabels:
+      app: prometheus-node-exporter
+      release: monitoring
+
+EOF
+  )
+  echo "$monitor" > monitor.yaml
+
+  daemonset=$(cat <<"EOF"
+apiVersion: apps/v1
+kind: DaemonSet
```

Comment:
We have the entire DaemonSet definition and it's hard to check what is the additional configuration. Can we describe in a comment above the newly added init containers and briefly what they do? **Note:** we have this file also in our config repo and test-infra repo, we can add this comment there as well.

Reply:
Nice suggestion! Some comments are added. 
In regards to the Istio sidecar injection, if that's what you're referring to, this cannot be accomplished because the `node-exporter` application uses the host's network which in turn tells Istio to skip injecting an Envoy sidecar proxy.